### ISSUE TITLE
Error connecting during stresstest

### ISSUE THREAD

**Issue by nicofff** [socketio/socket.io#2521]
I'm doing a stress test of an application that is using socket.io, and I'm seeing the clients getting disconnects after I've connected a couple hundred clients. 

The error I get is 

> [Error: xhr poll error] type: 'TransportError', description: 503 

I've managed to create a repro case: 

Server:

```

var app = require('http').createServer(function(){});
var io = require('socket.io')(app);
app.listen(3000);


```

Client:

```
var numClients = 10000
var socketclient = require('socket.io-client');

for (var i = 0; i <= numClients; i++) {
    var socket = socketclient("http://localhost:3000",{forceNew: true});
    socket.on('connect',function(){
        console.log("connected");
    });

    socket.on('error',function(err){
        console.log("error",err);
    });

    socket.on('disconnect',function(){
        console.log("disconnected");
    });
}

```

Forcing the transport to websockets only does seem to prevent the problem. But that's not something I'm willing to do on the application, and I'd like the stresstest to be as close to reality as posible.

**Comment by nicofff** [comment]
I finally got the stress test done. I had to include a 50ms delay between each new client connection across 4 socket.io servers (so more like a 200ms delay from the server perspective).

Aside from making stress test take longer than it should (I had an 1 hour ramp up to hit my 60k concurrent connections target), I see a potential problem if any of the servers gets restarted during peak hours, and having a rush of reconnections which might take longer than usual to get properly established due to this error, if they don't fail altogether.
